pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1261,6 +1261,39 @@ func (qjm *XController) ScheduleNext() {
 
 			if aggqj.LessEqual(resources) && qjm.nodeChecks(qjm.cache.GetUnallocatedHistograms(), qj) {
 				//Now evaluate quota
+				//Quota tree design:
+				// Quota tree should have two nodes : default node and experimental node
+				// default node should have the same quota as root node with hardlimit set to True
+				// experimental node is the node which will borrow resources with hardlimit set to False but
+				// quota set to much lower value than default node.
+
+				// All non-quota submission AWs will consume quota from default node.
+
+				//Find if AW has label named "quota_context"
+				allLabels := qj.DeepCopy().Labels
+				quotaSetForAw := false
+				for labelk, _ := range allLabels {
+					if labelk == "quota_context" {
+						quotaSetForAw = true
+					}
+				}
+				//When label "quota_context" not found add it
+				//this will only work for quota tree and not quota forest
+				if quotaSetForAw == false {
+					if len(allLabels) == 0 {
+						allLabels = make(map[string]string)
+					}
+					allLabels["quota_context"] = "default"
+					qj.SetLabels(allLabels)
+					if err := qjm.updateEtcd(qj, "ScheduleNext - setDefaultQuota"); err == nil {
+						klog.V(3).Infof("[ScheduleNext] Default quota added to AW %v", qj.Name)
+					}
+
+					if err != nil {
+						klog.V(3).Infof("[ScheduleNext] Failed to added default quota to AW %v, skipping dispatch of AW", qj.Name)
+						return
+					}
+				}
```

Comment:
Why not pull out this code as private method on the `controller` struct or as helper method? If you have a